pkg/deploy/lattice/access_log_subscription_manager.go

```diff
@@ -0,0 +1,106 @@
+}
+
+type defaultAccessLogSubscriptionManager struct {
+	log   gwlog.Logger
+	cloud aws.Cloud
+}
+
+func NewAccessLogSubscriptionManager(
+	log gwlog.Logger,
+	cloud aws.Cloud,
+) *defaultAccessLogSubscriptionManager {
+	return &defaultAccessLogSubscriptionManager{
+		log:   log,
+		cloud: cloud,
+	}
+}
+
+func (m *defaultAccessLogSubscriptionManager) Create(
+	ctx context.Context,
+	accessLogSubscription *lattice.AccessLogSubscription,
+) (*lattice.AccessLogSubscriptionStatus, error) {
+	vpcLatticeSess := m.cloud.Lattice()
+
+	var resourceIdentifier string
+	if accessLogSubscription.Spec.SourceType == lattice.ServiceNetworkSourceType {
+		serviceNetwork, err := getServiceNetworkWithName(ctx, vpcLatticeSess, accessLogSubscription.Spec.SourceName)
+		if err != nil {
+			return nil, err
+		}
+		resourceIdentifier = *serviceNetwork.Arn
+	} else {
+		service, err := getServiceWithName(ctx, vpcLatticeSess, accessLogSubscription.Spec.SourceName)
+		if err != nil {
+			return nil, err
+		}
+		resourceIdentifier = *service.Arn
+	}
+
+	createALSInput := &vpclattice.CreateAccessLogSubscriptionInput{
+		ResourceIdentifier: &resourceIdentifier,
+		DestinationArn:     &accessLogSubscription.Spec.DestinationArn,
+	}
+
+	createALSOutput, err := vpcLatticeSess.CreateAccessLogSubscriptionWithContext(ctx, createALSInput)
+	if err != nil {
+		if e, ok := err.(*vpclattice.ConflictException); ok {
+			return nil, services.NewConflictError(string(accessLogSubscription.Spec.SourceType), accessLogSubscription.Spec.SourceName, e.Message())
+		} else if e, ok := err.(*vpclattice.AccessDeniedException); ok {
+			return nil, services.NewInvalidError(e.Message())
+		} else if e, ok := err.(*vpclattice.ResourceNotFoundException); ok {
+			if *e.ResourceType == "SERVICE_NETWORK" || *e.ResourceType == "SERVICE" {
+				return nil, services.NewNotFoundError(string(accessLogSubscription.Spec.SourceType), accessLogSubscription.Spec.SourceName)
+			}
+			return nil, services.NewInvalidError(e.Message())
+		}
+		return nil, err
+	}
+
+	return &lattice.AccessLogSubscriptionStatus{
+		Arn: *createALSOutput.Arn,
+		Id:  *createALSOutput.Id,
+	}, nil
+}
+
+func getServiceNetworkWithName(ctx context.Context, vpcLatticeSess services.Lattice, name string) (*vpclattice.ServiceNetworkSummary, error) {
+	serviceNetworkSummaries, err := vpcLatticeSess.ListServiceNetworksAsList(ctx, &vpclattice.ListServiceNetworksInput{})
+	if err != nil {
+		return nil, err
+	}
+	for _, serviceNetworkSummary := range serviceNetworkSummaries {
+		if *serviceNetworkSummary.Name == name {
+			return serviceNetworkSummary, nil
+		}
+	}
+	return nil, services.NewNotFoundError("ServiceNetwork", name)
+}
+
+func getServiceWithName(ctx context.Context, vpcLatticeSess services.Lattice, name string) (*vpclattice.ServiceSummary, error) {
+	serviceSummaries, err := vpcLatticeSess.ListServicesAsList(ctx, &vpclattice.ListServicesInput{})
```

Comment:
We have a FindService()  function, same with that one?
https://github.com/aws/aws-application-networking-k8s/blob/6cccaa51c8bd797a0576302bc4bb3757604c94fb/pkg/aws/services/vpclattice.go#L231

Reply:
Same as above!